areas/deploy/src/AzureMcp.Deploy/Services/Util/AzdResourceLogService.cs

```diff
@@ -0,0 +1,108 @@
+using System.Diagnostics.CodeAnalysis;
+using Azure.Core;
+using YamlDotNet.Serialization;
+
+namespace Areas.Deploy.Services.Util;
+
+public static class AzdResourceLogService
+{
+    private const string AzureYamlFileName = "azure.yaml";
+
+    [RequiresDynamicCode("Uses YamlDotNet for deserialization.")]
+    public static async Task<string> GetAzdResourceLogsAsync(
```

Comment:
The RequiresDynamicCode annotation signifies that the associated API is not compatible with Ahead-of-Time (AOT) compilation. All APIs utilized within the Azure MCP server must fully compliant with both AOT and trimming requirements. In the coming weeks, we plan to ship a native binary for the Azure MCP server for all arch, which will not be possible if non-compliant APIs are present.